src/Core/Extractors/nNASOS.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace ArchiveCacheManager
+{
+    // this extractor is specificilly for nNASOS1.8 available at https://archive.org/details/nNASOS1.8
+    public class NNASOS : Extractor
+    {
+        // Needed to place in LanchBox root ThirdParty/ folder as nNASOS uses a .dll itself that LaunchBox
+        // mistakenly tries to register as a plugin, crashing LaunchBox on startup.
+        string executablePath = Path.Combine(PathUtils.GetLaunchBoxNNASOSRootPath(), "nNASOS.exe");
+
+        public NNASOS()
+        {
+
+        }
+
+        public static bool SupportedType(string archivePath)
+        {
+            return PathUtils.HasExtension(archivePath, new string[] { ".7z", ".zip", ".iso.dec" });
```

Comment:
`PathUtils.HasExtension()` calls `Path.GetExtension()` which will return .dec for .iso.dec files
```suggestion
            return PathUtils.HasExtension(archivePath, new string[] { ".7z", ".zip", ".dec" });
```